### PR TITLE
added Ninja Multi-Config to regexp

### LINKED ISSUE #18490: CMake: Potential bug (code review)
Disclaimer: not tested, just reviewing the code...

Since CMake now provide a [multi configuration ninja generator](https://cmake.org/cmake/help/latest/generator/Ninja%20Multi-Config.html), I guess this check is wrong:
https://github.com/opencv/opencv/blob/a3e7c2d8e3ebb32a5ee58c8f3265dacead318572/CMakeLists.txt#L1032-L1034

Fix: please add `Ninja Multi-Config` (or simply `Multi-Config` :thinking:)  to the regexp...

### PR DESCRIPTION
**Pull request by krush11**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

fixes #18490

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -603,7 +603,7 @@ endif()
 # ----------------------------------------------------------------------------
 # In case of Makefiles if the user does not setup CMAKE_BUILD_TYPE, assume it's Release:
 if(CMAKE_GENERATOR MATCHES "Makefiles|Ninja" AND "${CMAKE_BUILD_TYPE}" STREQUAL "")
-  set(CMAKE_BUILD_TYPE Release)
+  set(CMAKE_DEFAULT_BUILD_TYPE Release)
```
Comment (alalek) [inline CMakeLists.txt:606]: > CMAKE_DEFAULT_BUILD_TYPE
> New in version 3.17.

https://cmake.org/cmake/help/latest/variable/CMAKE_DEFAULT_BUILD_TYPE.html

1. We must do not break support for older CMake.
2. What is about `try_compile()` checks?

**Review by alalek — COMMENTED** [review]
> added Ninja Multi-Config to regexp
> regexp

I do not see any changes here in regexp.

File: CMakeLists.txt (hunk 1 above)
Comment (krush11) [reply CMakeLists.txt]: > > added Ninja Multi-Config to regexp
> > regexp
> 
> I do not see any changes here in regexp.

i did not understand what changes should have been made for `ninja multi-config` so i updated as much as i was confident about.
If you could direct me on what changes are needed i would be more than happy to do them

**Comment by mshabunin** [comment]
@krush11 , I tested current _master_ branch with Ninja Multi-Config generator and didn't find any problems, please file a new issue if you experience one.

**Comment by alalek** [comment]
Issue suggests to update used regex condition: `CMAKE_GENERATOR MATCHES "Xcode|Visual Studio"`
But this PR is going in a different strange way.

**Pull request closed without merging**